x/auction/keeper/querier.go

```diff
@@ -1,151 +1,147 @@
+	nextAuctionID, _ := keeper.GetNextAuctionID(ctx)
 
-	// Encode results
-	bz, err := codec.MarshalJSONIndent(keeper.cdc, params)
+	bz, err := legacyQuerierCdc.MarshalJSON(nextAuctionID)
 	if err != nil {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
 	}
-
 	return bz, nil
 }
 
 // filterAuctions retrieves auctions filtered by a given set of params.
 // If no filters are provided, all auctions will be returned in paginated form.
-func filterAuctions(ctx sdk.Context, auctions types.Auctions, params types.QueryAllAuctionParams) types.Auctions {
-	filteredAuctions := make(types.Auctions, 0, len(auctions))
-
+func filterAuctions(ctx sdk.Context, auctions []types.Auction, params types.QueryAllAuctionParams, legacyQuerierCdc *codec.LegacyAmino) []types.Auction {
+	filteredAuctions := make([]types.Auction, 0, len(auctions))
 	for _, auc := range auctions {
-		matchType, matchOwner, matchDenom, matchPhase := true, true, true, true
-
-		// match auction type (if supplied)
-		if len(params.Type) > 0 {
-			matchType = auc.GetType() == params.Type
-		}
-
-		// match auction owner (if supplied)
-		if len(params.Owner) > 0 {
-			if cAuc, ok := auc.(types.CollateralAuction); ok {
-				foundOwnerAddr := false
-				for _, addr := range cAuc.GetLotReturns().Addresses {
-					if addr.Equals(params.Owner) {
-						foundOwnerAddr = true
-						break
-					}
-				}
-				if !foundOwnerAddr {
-					matchOwner = false
-				}
-			}
+		isMatch := auctionIsMatch(auc, params)
+		if isMatch {
+			filteredAuctions = append(filteredAuctions, auc)
 		}
+	}
+	return filteredAuctions
+}
 
-		// match auction denom (if supplied)
-		if len(params.Denom) > 0 {
-			matchDenom = auc.GetBid().Denom == params.Denom || auc.GetLot().Denom == params.Denom
-		}
+func auctionIsMatch(auc types.Auction, params types.QueryAllAuctionParams) bool {
+	matchType, matchOwner, matchDenom, matchPhase := true, true, true, true
 
-		// match auction phase (if supplied)
-		if len(params.Phase) > 0 {
-			matchPhase = auc.GetPhase() == params.Phase
-		}
+	// match auction type (if supplied)
+	if len(params.Type) > 0 {
+		matchType = auc.GetType() == params.Type
+	}
 
-		if matchType && matchOwner && matchDenom && matchPhase {
-			filteredAuctions = append(filteredAuctions, auc)
+	// match auction owner (if supplied)
+	if len(params.Owner) > 0 {
+		if cAuc, ok := auc.(*types.CollateralAuction); ok {
+			foundOwnerAddr := false
+			for _, addr := range cAuc.GetLotReturns().Addresses {
+				if addr == params.Owner.String() {
+					foundOwnerAddr = true
+					break
+				}
+			}
+			if !foundOwnerAddr {
+				matchOwner = false
+			}
 		}
 	}
 
-	start, end := client.Paginate(len(filteredAuctions), params.Page, params.Limit, 100)
-	if start < 0 || end < 0 {
-		filteredAuctions = types.Auctions{}
-	} else {
-		filteredAuctions = filteredAuctions[start:end]
```

Comment:
looks like the pagination was dropped. I'd add it back to avoid changes to the querier. Otherwise there'll be more work to do in updating bots/services.

Reply:
Updated in https://github.com/Kava-Labs/kava/pull/1058/commits/a61fb18b4a29fcbc6a125c78b4fdbc9af1076047